### PR TITLE
syslog-ng custom log metrics

### PR DESCRIPTION
**Pull request by pepov**
SyslogNG custom source and output metrics

Source metrics are put after the json parser.
```
kind: Logging
apiVersion: logging.banzaicloud.io/v1beta1
metadata:
  name: logging
spec:
  controlNamespace: default
  fluentbit: {}
  syslogNG:
    metrics: {}
    sourceMetrics:
      - key: custom_input
        labels:
          test: asd
```
Results in:
```
source "main_input" {
    channel {
        source {
            network(flags("no-parse") port(601) transport("tcp") max-connections(100) log-iw-size(10000));
        };
        parser {
            json-parser(prefix("json."));
            metrics-probe(key("custom_input") labels(
                "logging" => "logging"
                "test" => "asd"
            ));
        };
    };
};
```

Output metrics are put before the log reaches the destination and is decorated with the output metadata like: `name`, `namespace` and `scope` which holds whether the output is a `local` or `global` one.
```
apiVersion: logging.banzaicloud.io/v1beta1
kind: SyslogNGFlow
metadata:
  name: all1
spec:
  match: {}
  outputMetrics:
    - key: custom_output
      labels:
        flow: all1
  localOutputRefs:
    - http
  globalOutputRefs:
    - http2
```
Becomes:
```
filter "flow_default_all1_ns_filter" {
    match("default" value("json.kubernetes.namespace_name") type("string"));
};
log {
    source("main_input");
    filter("flow_default_all1_ns_filter");
    log {
        parser {
            metrics-probe(key("custom_output") labels(
                "flow" => "all1"
                "logging" => "logging"
                "output_name" => "http2"
                "output_namespace" => "default"
                "output_scope" => "global"
            ));
        };
        destination("clusteroutput_default_http2");
    };
    log {
        parser {
            metrics-probe(key("custom_output") labels(
                "flow" => "all1"
                "logging" => "logging"
                "output_name" => "http"
                "output_namespace" => "default"
                "output_scope" => "local"
            ));
        };
        destination("output_default_http");
    };
};
```

Metrics:
```
k curl logging-syslog-ng-0:9577/metrics  | grep custom_
# TYPE syslogng_custom_output gauge
syslogng_custom_output{flow="all1",logging="logging",output_name="http2",output_namespace="default",output_scope="global"} 42
syslogng_custom_output{flow="all1",logging="logging",output_name="http",output_namespace="default",output_scope="local"} 42
syslogng_custom_output{flow="all2",logging="logging",output_name="http2",output_namespace="default",output_scope="global"} 154
# TYPE syslogng_custom_input gauge
syslogng_custom_input{logging="logging"} 154
```

### CODE REVIEW

**Review by tarokkk — APPROVED** [review]

File: pkg/sdk/logging/model/syslogng/config/config.go [hunk 1]
```diff
@@ -112,6 +112,21 @@ func configRenderer(in Input) (render.Renderer, error) {
 		in.Logging.Spec.SyslogNGSpec.JSONKeyPrefix = "json" + keyDelim(in.Logging.Spec.SyslogNGSpec.JSONKeyDelimiter)
 	}
 
+	parserDrivers := []render.Renderer{
```
Comment (siliconbrain) [inline pkg/sdk/logging/model/syslogng/config/config.go:115]: I'd name this something more specific to better convey its place
```suggestion
	sourceParsers := []render.Renderer{
```

**Review by siliconbrain — APPROVED** [review]
lgtm

**Review by OverOrion — APPROVED** [review]

File: pkg/sdk/logging/model/syslogng/config/config.go (hunk 1 above)
Comment (pepov) [reply pkg/sdk/logging/model/syslogng/config/config.go]: 3f31b3d77ca1aa50bb0403e04cd8e4613c30b24f

**Comment by pepov** [comment]
Update: based on @tarokkk's request logging name is now part of the automatic metric labels

**Pull request merged**